bigquery/client.py

```diff
@@ -166,12 +181,82 @@ def query(self, query, max_results=None, timeout=0, dry_run=False):
 
         # raise exceptions if it's not an async query
         # and job is not completed after timeout
-        if not job_complete and timeout:
+        if not job_complete and query_data.get("timeoutMs", False):
             logging.error('BigQuery job %s timeout' % job_id)
             raise BigQueryTimeoutException()
 
         return job_id, [self._transform_row(row, schema) for row in rows]
 
+    def query(self, query, max_results=None, timeout=0, dry_run=False):
+        """Submit a query to BigQuery.
+
+        Args:
+            query: BigQuery query string.
+            max_results: maximum number of rows to return per page of results.
+            timeout: how long to wait for the query to complete, in seconds,
+                     before the request times out and returns.
+            dry_run: if True, the query isn't actually run. A valid query will
+                     return an empty response, while an invalid one will return
+                     the same error message it would if it wasn't a dry run.
+
+        Returns:
+            job id and query results if query completed. If dry_run is True,
+            job id will be None and results will be empty if the query is valid
+            or a dict containing the response if invalid.
+
+        Raises:
+            BigQueryTimeoutException on timeout
+        """
+
+        logging.debug('Executing query: %s' % query)
+
+        query_data = {
+            'query': query,
+            'timeoutMs': timeout * 1000,
+            'dryRun': dry_run,
+            'maxResults': max_results,
+        }
+        return self._submit_query_job(query_data)
+
+    def save_query_as_table(self, query, dataset=None, table_name=None,
+        write_disposition="WRITE_TRUNCATE", timeout=0):
+        """ Submit a query to BigQuery and save results to a table.
+
+        Args:
+            query: SQL to run
+            dataset: destination dataset for query results
+            table_name: destination table for query results
+            write_disposition: one of the following string values:
+                               "WRITE_TRUNCATE" -  (default) overwrites all existing rows,
+                                                   removing previous data in the table.
+                               "WRITE_APPEND"   -  adds rows to the destination table.
+                               "WRITE_EMPTY"    -  only saves its results to the destination
```

Comment:
Yeah, not sure about what makes a better API.  I'm between keeping the old mode names or trying to pick something a little more informative:

```
OVERWRITE = "WRITE_TRUNCATE"
APPEND = "WRITE_APPEND"
IF_EMPTY = "WRITE_EMPTY"
```

versus

```
TRUNCATE = "WRITE_TRUNCATE"
APPEND = "WRITE_APPEND"
EMPTY = "WRITE_EMPTY"
```

Constants to the user would look something like
```
client.save_query_to_table(
    "select country, count(*) ct from [d.fruits] group by country",
    dataset="tmp",
    table_name="something",
    write_disposition=client.IF_EMPTY
)
```